packages/svelteui-core/src/components/JsonInput/JsonInput.svelte

```diff
@@ -0,0 +1,53 @@
+<script lang="ts">
+	import type { JsonInputProps as $$JsonInputProps } from './JsonInput';
+	import Textarea from '../Textarea/Textarea.svelte';
+	import { validateJson } from './validate-json/validate-json';
+
+	interface $$Props extends $$JsonInputProps {}
+
+	export let formatOnBlur: $$Props['formatOnBlur'] = false,
+		error: $$Props['error'] = null,
+		validationError: $$Props['validationError'] = null,
+		onBlur: $$Props['on:blur'] = undefined,
+		readOnly: $$Props['readonly'] = undefined,
+		value: $$Props['value'] = undefined,
+		serialize: $$Props['serialize'] = JSON.stringify,
+		deserialize: $$Props['deserialize'] = JSON.parse;
+
+	let _value = value;
+	let valid = validateJson(_value, deserialize);
+
+	const handleBlur = (event) => {
+		onBlur?.(event);
+		valid = validateJson(event.currentTarget.value, deserialize);
+		if (formatOnBlur && !readOnly && valid && event.currentTarget.value.trim() !== '') {
+			_value = serialize(deserialize(event.currentTarget.value), null, 2);
+		}
+	};
+</script>
+
+<!--
+@component
+
+Textarea specially made for JSON input.
+
+@see https://svelteui.org/core/json-input
+@example
+    ```tsx
+    <JsonInput
+      label='Data'
+      placeholder='Enter JSON data'
+      required
+    />
+    ```
+-->
+
+<Textarea
```

Comment:
Hum, I would increase the default value of rows to 4 or 6, since the default value of 2 is too small for a JSON input. This should also be possible to override.

Reply:
Cool. increased to 4. feels like the sweet spot for a default